src/std/net/json-rpc.ss

```diff
@@ -126,22 +106,121 @@
+                headers: `(("Content-Type" . "application/json-rpc")
+                           ;; ("Accept" . "application/json-rpc, application/json, application/jsonrequest")
+                           ,@headers)
                 cookies: cookies
                 data: (string->bytes request-string))))
   (def response-json
     (try (content->json response-bytes) ;; todo: move to decode-json-rpc-response ?
          (catch (e) (raise (malformed-response request-id: id response: response-bytes e: e)))))
   (when log
     (log [from: server-url response: (bytes->string response-bytes)]))
-
   (decode-json-rpc-response
    result-decoder
    (and (hash-table? response-json) (hash-get response-json "id"))
    response-json))
+
+(def (request->string method params id)
+  (try (json-object->string
+        (json-rpc-request jsonrpc: json-rpc-version method: method params: params id: id))
+       (catch (e) (raise (malformed-request method: method params: params e: (error-message e))))))
+
+(def (decode-json-rpc-response decoder request-id response-json)
+  (def (mal e)
+    (raise (malformed-response request-id: request-id response: response-json e: (error-message e))))
+  (def response (with-catch mal (cut trivial-json-object->class json-rpc-response::t response-json)))
+  (def jsonrpc (@ response jsonrpc))
+  (def result (@ response result))
+  (def error (@ response error))
+  (def id (@ response id))
+  (unless (or (void? jsonrpc) ;; a 1.0 server might fail to include this field
+              (equal? jsonrpc json-rpc-version)) ;; but a recent server must reply with same version
+    (mal "bad json_rpc_version"))
+  (unless (or (void? result) (void? error))
+    (mal "result error conflict"))
+  (unless (equal? id request-id)
+    (mal "bad id"))
+  (if (void? error)
+    (with-catch mal (cut decoder result))
+    (raise (with-catch mal (cut json->json-rpc-error error)))))
+
+;;; Server code
+
+;; Acceptable content-types for json-rpc requests and responses
+(def +json-rpc-content-types+
+  '(("application/json-rpc" ;; SHOULD be that
+     "application/json" "application/jsonrequest"))) ;; MAY be these
+
+;; http handler for json-rpc
+;; NB: Be sure to wrap into a condition-handler, or some such.
+(def (json-rpc-handler processor)
+  (lambda (req res)
+    (let/cc return
+      (def (bad-request)
+        (http-response-write-condition res (Bad-Request))
+        (return))
+      ;; Skip these checks specified in JSON RPC over HTTP
+      #;(let (headers (http-request-headers req))
+          (unless (and (member (assget "Content-Type" headers) +json-rpc-content-types+)
+                       (assget "Content-Length" headers)
+                       (member (assget "Accept" headers) +json-rpc-content-types+))
+            (bad-request)))
+      (case (http-request-method req)
+        ;; TODO: implement GET as per https://www.jsonrpc.org/historical/json-rpc-over-http.html
+        #;((GET) (json-rpc-handler/GET req res processor))
```

Comment:
let's implement that.

Reply:
Done. It was probably more effort than is worth, considering the strictures on using it and the lack of support in general.